gui/dialogs/mcduweb.xml

```diff
@@ -0,0 +1,137 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<!-- Copyright (c) 2020 Josh Davidson (Octal450) -->
+
+<!-- A3XX MCDU web launcher Dialog -->
+
+<PropertyList>
+
+	<name>mcdu-web</name>
+	<layout>vbox</layout>
+	<pref-width>480</pref-width>
+
+    <group>
+        <layout>hbox</layout>
+        <text>
+            <halign>left</halign>
+            <label>MCDU web on browser/tablet/smartphone</label>
+        </text>
+
+        <button>
+            <halign>right</halign>
+            <pref-width>20</pref-width>
+            <pref-height>20</pref-height>
+            <legend>X</legend>
+            <key>Esc</key>
+            <binding>
+                <command>dialog-close</command>
+            </binding>
+        </button>
+    </group>
+	
+    <hrule/>
+	
+	<group>
+		<layout>vbox</layout>
+		
+		<text>
+			<halign>left</halign>
+			<label>Please check: HTTP service running is required.</label>
+			<visible>
+				<equals>
+					<property>/sim/http/running</property>
+					<value>0</value>
+				</equals>
+			</visible>			
+		</text>
+		<text>		
+			<halign>left</halign>
+			<label>Start FlightGear with -httpd=8080</label>
+			<visible>
+				<equals>
+					<property>/sim/http/running</property>
+					<value>0</value>
+				</equals>
+			</visible>			
+		</text>
+
+    </group>
+	
+	<group>
+		<layout>vbox</layout>
+
+		<text>
+			<halign>center</halign>
+			<label>Open MCDU on browser</label>
+		</text>
+
+		<button>
+			<halign>center</halign>
+			<legend>launch browser</legend>
+			<binding>
+				<command>nasal</command>
+				<script>
+					var n = getprop("/sim/http/running",0);
+					if( n != 1 ) {
+					  gui.popupTip("Internal webserver not running. Restart FlightGear with -httpd=8080", 5.0);
+					} else {
+					  var _url = "http://localhost:" ~ getprop("/sim/http/options/listening-port") ~ "/aircraft-dir/WebPanel/mcdu.html";
+					  fgcommand("open-browser", props.Node.new({ "url": _url }));
+					}				
+				</script>
+			</binding>
+		</button>
+	</group>	
+	
+	<group>
+		<layout>vbox</layout>
+		<text>
+			<halign>center</halign>
+			<label></label>
+		</text>
+		<text>
+			<halign>center</halign>
+			<label>Scan QR code with your smartphone or tablet</label>
+		</text>
+
+	</group>
+		
+	<canvas>
+		<name>mcduwebqr</name>
+		<valign>center</valign>
+		<halign>center</halign>
+		<stretch>false</stretch>
+		<pref-width>320</pref-width>
+		<pref-height>320</pref-height>
+		<nasal>
+			<load>
+				<![CDATA[
+				    var n = props.globals.getNode("/sim/http/running");
+					var mcdu_canvas_dlg = canvas.get(cmdarg());
+					var root = mcdu_canvas_dlg.createGroup();
+					mcdu_canvas_dlg.setColorBackground(1, 1, 1, 1);
+					var _url = "";
+					if( getprop("sim/http/running",0) == 1 ) _url = "http://localhost:" ~ getprop("/sim/http/options/listening-port") ~ "/aircraft-dir/WebPanel/mcdu.html";
```

Comment:
This won't work!
You're referring to localhost but ask the user to to scan it with a phone/tablet-> different device.
You'd need to use either the IP address or the host name for this to work.

Reply:
Only work if you are on the same network, your smartphone/tablet it suppose to use your home wifi connection.
It needs to clarify on doc.
It's possible to expose on external network, it's a bit complex but it's out of flightgear context.

Reply:
After meditation... you are right. I need to detect local IP to let load mcdu on external devices.

Reply:
@merspieler @legoboyvdlp I have update remote mcdu with input for local-ip request and some minor fixes.